bundles/org.openhab.binding.upnpcontrol/src/main/java/org/openhab/binding/upnpcontrol/internal/handler/UpnpRendererHandler.java

```diff
@@ -0,0 +1,868 @@
+        }
+    }
+
+    @Override
+    public void dispose() {
+        cancelSubscriptionRefreshJob();
+        removeSubscription("AVTransport");
+
+        cancelTrackPositionRefresh();
+
+        super.dispose();
+    }
+
+    private void cancelSubscriptionRefreshJob() {
+        ScheduledFuture<?> refreshJob = subscriptionRefreshJob;
+
+        if (refreshJob != null) {
+            refreshJob.cancel(true);
+        }
+        subscriptionRefreshJob = null;
+
+        upnpSubscribed = false;
+    }
+
+    private void initRenderer() {
+        if (!upnpSubscribed) {
+            addSubscription("AVTransport", SUBSCRIPTION_DURATION_SECONDS);
+            upnpSubscribed = true;
+
+            subscriptionRefreshJob = scheduler.scheduleWithFixedDelay(subscriptionRefresh,
+                    SUBSCRIPTION_DURATION_SECONDS / 2, SUBSCRIPTION_DURATION_SECONDS / 2, TimeUnit.SECONDS);
+        }
+        getProtocolInfo();
+        getTransportState();
+
+        updateStatus(ThingStatus.ONLINE);
+    }
+
+    /**
+     * Invoke Stop on UPnP AV Transport.
+     */
+    public void stop() {
+        Map<String, String> inputs = Collections.singletonMap("InstanceID", Integer.toString(avTransportId));
+
+        invokeAction("AVTransport", "Stop", inputs);
+    }
+
+    /**
+     * Invoke Play on UPnP AV Transport.
+     */
+    public void play() {
+        Map<String, String> inputs = new HashMap<>();
+        inputs.put("InstanceID", Integer.toString(avTransportId));
+        inputs.put("Speed", "1");
+
+        invokeAction("AVTransport", "Play", inputs);
+    }
+
+    /**
+     * Invoke Pause on UPnP AV Transport.
+     */
+    public void pause() {
+        Map<String, String> inputs = Collections.singletonMap("InstanceID", Integer.toString(avTransportId));
+
+        invokeAction("AVTransport", "Pause", inputs);
+    }
+
+    /**
+     * Invoke Next on UPnP AV Transport.
+     */
+    protected void next() {
+        Map<String, String> inputs = Collections.singletonMap("InstanceID", Integer.toString(avTransportId));
+
+        invokeAction("AVTransport", "Next", inputs);
+    }
+
+    /**
+     * Invoke Previous on UPnP AV Transport.
+     */
+    protected void previous() {
+        Map<String, String> inputs = Collections.singletonMap("InstanceID", Integer.toString(avTransportId));
+
+        invokeAction("AVTransport", "Previous", inputs);
+    }
+
+    /**
+     * Invoke SetAVTransportURI on UPnP AV Transport.
+     *
+     * @param URI
+     * @param URIMetaData
+     */
+    public void setCurrentURI(String URI, String URIMetaData) {
+        Map<String, String> inputs = new HashMap<>();
+        try {
+            inputs.put("InstanceID", Integer.toString(avTransportId));
+            inputs.put("CurrentURI", URI);
+            inputs.put("CurrentURIMetaData", URIMetaData);
+        } catch (NumberFormatException ex) {
+            logger.error("Action Invalid Value Format Exception {}", ex.getMessage());
```

Comment:
These should probably not be logged to error as it indicates a problem with the data, not specifically with the binding. Error logging in openHAB is intended for severe errors.

When the exception would be thrown, dioes it still make sense to call invokeAction? otherwise the call (and map) should be inside the try-catch.